src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/Microsoft.WinFX.targets

```diff
@@ -377,18 +380,22 @@
   <PropertyGroup>
 
        <MarkupCompilePass2ForMainAssemblyDependsOn>
+                     CreateTemporaryTargetAssemblyProject;
                      GenerateTemporaryTargetAssembly;
+                     GenerateTemporaryTargetAssembly2;
```

Comment:
Good catch. This came as a pre 5.0 change initially and rules like "don't introduce publics unless unavoidable" went into a blindspot for previous reviewers like me... fresh reviewers are great for these.

Reply:
Thanks.  One possible solution is to use the GenerateTemporaryTargetAssembly target as the entry point for both the old and new behavior, so applications hooking GenerateTemporaryTargetAssembly continue to work.  Any objections to that approach?  

For the consolidation in 6.0, there is already discussion of removing the temp project generation and compilation entirely and instead using source generators during the main application compilation.  But if we do keep the temp project, the idea would be to move the new behavior in to the GTTA target and GTTA task.  The GTTA task would need to create the temporary project, copy and rename the parent project's generate NuGet props/targets, and compile the temporary project.  WPF applications that call the public GTTA task directly should continue to work.  In other words, all the logic from the CreateTemporaryTargetASsemblyProject task and CompileTemporaryTargetAssemblyProject task would be moved in to the GTTA task.

Reply:
I'll go ahead with moving everything under a common GenerateTemporaryTargetAssembly target.  We can roll it back if there are objections.  Thanks.

Reply:
As the two of you pointed out, there are scenarios that may break due to this change: 

1. Wpf application hooks the GTTA (GenerateTemporaryTargetAssembly) target
2. Wpf application calls the public GTTA target directly
3. Wpf applications calls the public GTTA task directly

To support 1), the GTTA target must always be called during the build; for 2), The GTTA target must exist with the old behavior; for 3), the GTTA task must remain as-is (old behavior).

These should continue to work when compiling existing applications without having to set properties explicitly.  

And for consolidation to a single task/target in .NET 6.0 (and to avoid breaking applications written between 5.0.2 and 6.0 by removing public targets/tasks) no new public targets/tasks should be created, if possible.   So the new behavior should placed directly in the the existing GTTA task or should be easily movable in to the existing GTTA task.

A possible change that supports the scenarios and allows for removal of the legacy behavior in .NET 6.0 is here: https://github.com/dotnet/wpf/commit/9e0a1f4d7f779f314b1c9acf46f2dd61c3c2b85e

The new tasks have been removed and everything has been moved in to GTTA.  If this seems reasonable I'll move it over to this PR.

Reply:
This is resolved with the most recent change.  Thanks.